src/Nethermind/Nethermind.Evm/Tracing/ITxTracer.cs

```diff
@@ -166,9 +166,10 @@ public interface ITxTracer : IWorldStateTracer, IDisposable
     /// <param name="gas"></param>
     /// <param name="opcode"></param>
     /// <param name="pc"></param>
+    /// <param name="executingAccount"></param>
     /// <param name="isPostMerge"></param>
     /// <remarks>Depends on <see cref="IsTracingInstructions"/></remarks>
-    void StartOperation(int depth, long gas, Instruction opcode, int pc, bool isPostMerge = false);
+    void StartOperation(int depth, long gas, Instruction opcode, int pc, Address executingAccount, bool isPostMerge = false);
```

Comment:
I spent the past day and a half investigating deriving the `executingAccount` in the tracer's `ReportAction`/`SetOperationStorage` methods without needing to modify the `ITxTracer` API, however the approach started really blowing up in scope and had a bunch of edge cases that I was having difficulty debugging. Since this PR is meant to optimize performance for the tracers, I felt that having the executing account passed direclty from the virtual machine made the most sense after struggling to derive the executing account in the tracer itself.

If any reviewer feels strongly about not passing through the executing account and instead deriving it in the tracer, would you be able to pair with me on debugging? I feel pretty blocked moving forward with this update.

For some additional information about my investigation, I calculated the initial executing account with 
```
private Address GetInitialExecutingAccount(Address from, Address? to, bool isContractCreation, bool isSystem)
{
    UInt256 nonce = isContractCreation ? _worldState!.GetNonce(from) : 0;
    return to ?? (isSystem
        ? from
        : ContractAddress.From(from, nonce > 0 ? nonce - 1 : nonce));
}
```

Then, for each modifying operation the executing account was recalculated. Here's a [gist](https://gist.github.com/natebeauregard/f49ec8ba4b1179bc7a570dec3d56e247) link with my attempted changes as well.

Reply:
Maybe let's do it differently, pass `ExecutionEnvironment` into `ReportAction`?

Reply:
I just added a commit to pass through the execution environment as a param in `StartOperation` as discussed in our daily meeting